mesontest.py

```diff
@@ -284,189 +284,211 @@ def print_stats(self, numlen, tests, name, result, i, logfile, jsonlogfile):
+                if len(lines) > 104:
+                    print('\n'.join(lines[0:4]))
                     print('--- Listing only the last 100 lines from a long log. ---')
-                    lines = lines[-99:]
+                    lines = lines[-100:]
                 for line in lines:
                     print(line)
-        print('Full log written to %s.' % logfilename)
-        return self.error_count
+
+    def doit(self):
+        if self.is_run:
+            raise RuntimeError('Test harness object can only be used once.')
+        if not os.path.isfile(self.datafile):
+            print('Test data file. Probably this means that you did not run this in the build directory.')
+            return 1
+        self.is_run = True
+        tests = self.get_tests()
+        if not tests:
+            return 0
+        self.run_tests(tests)
+        return self.fail_count
 
     def get_tests(self):
         with open(self.datafile, 'rb') as f:
             tests = pickle.load(f)
+
+        if not tests:
+            print('No tests defined.')
+            return []
+
+        if self.options.suite:
+            tests = [t for t in tests if self.options.suite in t.suite]
+
+        if self.options.args:
+            tests = [t for t in tests if t.name in self.options.args]
+
+        if not tests:
+            print('No suitable tests defined.')
+            return []
+
         for test in tests:
             test.rebuilt = False
 
         return tests
 
-    def run_tests(self, log_base):
-        logfile_base = os.path.join(self.options.wd, 'meson-logs', log_base)
+    def open_log_files(self):
+        if not self.options.logbase or self.options.verbose:
+            return (None, None, None, None)
+
+        logfile_base = os.path.join(self.options.wd, 'meson-logs', self.options.logbase)
+
         if self.options.wrapper is None:
-            wrap = []
             logfilename = logfile_base + '.txt'
-            jsonlogfilename = logfile_base+ '.json'
+            jsonlogfilename = logfile_base + '.json'
         else:
-            wrap = self.options.wrapper.split()
-            namebase = wrap[0]
+            namebase = self.options.wrapper.split()[0]
             logfilename = logfile_base + '-' + namebase.replace(' ', '_') + '.txt'
             jsonlogfilename = logfile_base + '-' + namebase.replace(' ', '_') + '.json'
-        tests = self.get_tests()
-        if len(tests) == 0:
-            print('No tests defined.')
-            return
-        numlen = len('%d' % len(tests))
-        executor = conc.ThreadPoolExecutor(max_workers=self.options.num_processes)
-        futures = []
-        filtered_tests = filter_tests(self.options.suite, tests)
-
-        jsonlogfile = None
-        logfile = None
+
+        jsonlogfile =  open(jsonlogfilename, 'w')
+        logfile = open(logfilename, 'w')
+
+        logfile.write('Log of Meson test suite run on %s.\n\n' %
+                    datetime.datetime.now().isoformat())
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->